frontend/tests/integration/components/transportation/threshold-calculation-test.js

```diff
@@ -0,0 +1,115 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { setupMirage } from "ember-cli-mirage/test-support";
+import { association } from 'ember-cli-mirage';
+
+module('Integration | Component | transportation/threshold-calculation', function(hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  let hasAllLandUses =
+    {
+      trafficZone: 2,
+      project: association({
+        totalUnits: 1000,
+        commercialLandUse: [
+          {
+            name: "Fast Food Restaurant",
+            type: "fast-food",
+            grossSqFt: 40
+          },
+          {
+            name: "Restaurant (not fast food)",
+            type: "restaurant",
+            grossSqFt: 50
+          },
+          {
+            name: "Office",
+            type: "office",
+            grossSqFt: 15
+          },
+          {
+            name: "Regional Retail",
+            type: "regional-retail",
+            grossSqFt: 35
+          },
+          {
+            name: "Local Retail",
+            type: "local-retail",
+            grossSqFt: 60
+          },
+        ],
+        communityFacilityLandUse: [
+          {
+            name: "General Community Facility",
+            type: "community-facility",
+            grossSqFt: 100
+          }
+        ],
+        parkingLandUse: [
+          {
+            name: "Garages",
+            type: "garages",
+            spaces: 70
+          },
+          {
+            name: "Lots",
+            type: "lots",
+            spaces: 100
+          }
+        ],
+      })
+    }
+
+  test('it renders', async function(assert) {
+    let analysisMirage = server.create('transportation-analysis', hasAllLandUses);
+
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'transportation-analysis' }
+    );
+    let analysis = await project.get('transportationAnalysis');
+
+    this.set('transportationModel', analysis);
+
+    await render(hbs`{{transportation/threshold-calculation analysis=transportationModel}}`);
+
+    // sumOfRatios = residentialUnitsRatio + officeSqFtRatio + regionalRetailSqFtRatio + localRetailSqFtRatio +
+    // restaurantSqFtRatio + communityFacilitySqFtRatio + offStreetParkingSpacesRatio
+    // sumOfRatios = 7.014399999999999, this is rounded to 7.01 in computed property `detailedAnalysisPopupText`
+    assert.ok(this.element.querySelector('[data-test-popup="detailed analysis"]').textContent.includes('7.01'));
+
+    // ## We are testing 7 computed properties in this test that are calculated using
+    // the transportation analysis model's computed property thresholdFor
+    // thresholdFor(type) { return this.get(`trafficZoneThresholds.${type}.zone${this.trafficZone}`); }
```

Comment:
nit: a little confusing to have the function pasted here. wasn't sure at first if it was meant to be in this test

Reply:
yeah please delete uncommented code